azure-mgmt-containerinstance/src/main/java/com/microsoft/azure/management/containerinstance/implementation/ContainerGroupImpl.java

```diff
@@ -0,0 +1,314 @@
+import com.microsoft.azure.management.containerinstance.ImageRegistryCredential;
+import com.microsoft.azure.management.containerinstance.OperatingSystemTypes;
+import com.microsoft.azure.management.containerinstance.Port;
+import com.microsoft.azure.management.containerinstance.Volume;
+import com.microsoft.azure.management.resources.fluentcore.arm.models.implementation.GroupableParentResourceImpl;
+import rx.Observable;
+import rx.functions.Func1;
+
+import java.util.ArrayList;
+import java.util.Collection;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+/**
+ * Implementation for ContainerGroup and its create interfaces.
+ */
+@LangDefinition
+public class ContainerGroupImpl
+    extends
+    GroupableParentResourceImpl<
+                ContainerGroup,
+                ContainerGroupInner,
+                ContainerGroupImpl,
+                ContainerInstanceManager>
+    implements ContainerGroup,
+    ContainerGroup.Definition {
+
+    private Map<String, Container> containers;
+    private Map<String, Volume> volumes;
+    private List<String> imageRegistryServers;
+    private int[] externalTcpPorts;
+    private int[] externalUdpPorts;
+
+    protected ContainerGroupImpl(String name, ContainerGroupInner innerObject, ContainerInstanceManager manager) {
+        super(name, innerObject, manager);
+    }
+
+    @Override
+    protected void beforeCreating() {
+    }
+
+    @Override
+    protected Observable<ContainerGroupInner> createInner() {
+        return this.manager().inner().containerGroups().createOrUpdateAsync(this.resourceGroupName(), this.name(), this.inner());
+    }
+
+    @Override
+    protected void afterCreating() {
+        initializeChildrenFromInner();
+    }
+
+    @Override
+    protected void initializeChildrenFromInner() {
+        // Getting the container instances
+        if (this.inner().containers() != null && this.inner().containers().size() > 0) {
+            this.containers = new HashMap<>();
+            for (Container containerInstance : this.inner().containers()) {
+                this.containers.put(containerInstance.name(), containerInstance);
+            }
+        }
+
+        // Getting the volumes
+        if (this.inner().volumes() != null && this.inner().volumes().size() > 0) {
+            this.volumes = new HashMap<>();
+            for (Volume volume : this.inner().volumes()) {
+                this.volumes.put(volume.name(), volume);
+            }
+        }
+
+        // Getting the private image registry servers
+        if (this.inner().imageRegistryCredentials() != null && this.inner().imageRegistryCredentials().size() > 0) {
+            this.imageRegistryServers = new ArrayList<>();
+            for (ImageRegistryCredential imageRegistry : this.inner().imageRegistryCredentials()) {
+                this.imageRegistryServers.add(imageRegistry.server());
+            }
+        }
```

Comment:
same comment about `this.imageRegistryServers` as above

Reply:
fixed to return null in that case.